### PR TITLE
feat: implement json lines processor

### PR DESCRIPTION
**Pull request by cpendery**
Closes https://github.com/guacsec/guac/issues/51

### CODE REVIEW

File: pkg/handler/processor/guesser/format_jsonlines_test.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+			{ "abc": "def"}
+			{ "abc": "def"}
+			{ "abc": "def"}
+		`),
+		expected: processor.FormatJSONLines,
+	}, {
+		name:     "invalid JSON Lines",
+		blob:     []byte(`"abc": "def"`),
+		expected: processor.FormatUnknown,
+	}}
```
[the remark this answers is not in the document]
Comment (cpendery) [reply pkg/handler/processor/guesser/format_jsonlines_test.go:55]: The spec site says: ["Each Line is a Valid JSON Value"](https://jsonlines.org/#each-line-is-a-valid-json-value), so I'd assume it doesn't allow 2 lines inbetween based on that

File: pkg/handler/processor/guesser/format_jsonlines.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+type jsonLinesFormatGuesser struct{}
+
+func (_ *jsonLinesFormatGuesser) GuessFormat(blob []byte) processor.FormatType {
+	lines := strings.Split(strings.TrimSpace(string(blob)), "\n")
+	for _, line := range lines {
+		if !json.Valid([]byte(strings.TrimSpace(line))) {
+			return processor.FormatUnknown
+		}
+	}
+	if len(lines) <= 1 {
```
[the remark this answers is not in the document]
Comment (cpendery) [reply pkg/handler/processor/guesser/format_jsonlines.go:34]: The length of lines can never be 0 as we give a separator value to the `strings.Split` method. I'll add a test case to ensure we get the expected results for this case.

citation: From `strings.Split's` godocs, ["If s does not contain sep and sep is not empty, Split returns a slice of length 1 whose only element is s."](https://pkg.go.dev/strings#Split),

**Comment by lumjjb** [comment]
>[Merge branch 'main' into feat-lines](https://github.com/guacsec/guac/pull/64/commits/95cb748dd485d25594bd3d4a39f5fc9c4410ae73)

can we rebase instead of having a merge commit, to maintain linear history?

File: pkg/handler/processor/processor.go [hunk 3]
```diff
@@ -63,8 +64,9 @@ type FormatType string
 
 // Format* is the enumerables of FormatType
 const (
-	FormatJSON    FormatType = "JSON"
-	FormatUnknown FormatType = "UNKNOWN"
+	FormatJSON      FormatType = "JSON"
+	FormatJSONLines FormatType = "JSONLINES"
```
Comment (lumjjb) [inline pkg/handler/processor/processor.go:68]: nit: JSON_LINES for readability

**Review by lumjjb — COMMENTED** [review]
minor nits and rebase and should be good to go!

**Review by lumjjb — APPROVED** [review]
lgtm

**Review by pxp928 — APPROVED** [review]
LGTM. @cpendery just needs a quick rebase and done. Thanks for this!

**Pull request merged**